### PR TITLE
Firebase massaging for iOs and Android

### PR DESCRIPTION
**Pull request by isipisi281**
Finaly done implementing massaging.

### CODE REVIEW

**Comment by isipisi281** [comment]
There is still one bug on iOS:
https://github.com/firebase/quickstart-cpp/issues/13

**Comment by Larpon** [comment]
Wow! Thanks a lot for your efforts @isipisi281 !!
I'll walk my way through it ASAP

File: qtfirebase_target.pri [hunk 1]
```diff
@@ -35,7 +38,13 @@ RESOURCES += \
     $$PWD/qtfirebase.qrc
 
 android: {
-    message( "QtFirebase Android base" )
+    # add services to xml file
+    #https://github.com/firebase/quickstart-cpp/blob/e8c20f678a06a28ebb73132abcd79d93b27622d9/messaging/testapp/AndroidManifest.xml
+
+    # add the onNewIntet function to your acivity
```
Comment (Larpon) [inline qtfirebase_target.pri]: `onNewIntet` = `onNewIntent`

File: fake/src/qtfirebaseauthenticate.h [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+#if defined(qFirebaseAuthenticate)
+#undef qFirebaseAuthenticate
+#endif
+#define qFirebaseAuthenticate (static_cast<QtFirebaseAuthenticate*>(QtFirebaseAuthenticate::instance()))
+
+#include <QObject>
+#include <QVariantMap>
+#include <QVariantList>
+
+class QtFirebaseAuthenticate : public QObject
```
Comment (Larpon) [inline fake/src/qtfirebaseauthenticate.h]: I can't see QtFirebaseAuthenticate being registered as a QML type anywhere - shouldn't it have it's own QML type?

File: qtfirebase_target.pri [hunk 3]
```diff
@@ -12,14 +12,17 @@ QML_IMPORT_PATH += $$PWD
 INCLUDEPATH += $$PWD
 
 INCLUDEPATH += \
-    $$QTFIREBASE_SDK_PATH/include
+    $$QTFIREBASE_SDK_PATH/include \
+    \
```
Comment (Larpon) [inline qtfirebase_target.pri]: ??? what does these do?

File: qtfirebase_target.pri [hunk 4]
```diff
@@ -44,91 +53,78 @@ android: {
...
-            -framework FirebaseAnalytics
-    }
-
-    HEADERS += $$PWD/src/qtfirebaseanalytics.h
-    SOURCES += $$PWD/src/qtfirebaseanalytics.cpp
-
-    LIBS += -L$$QTFIREBASE_SDK_LIBS_PATH -lanalytics
-}
+PRE_TARGETDEPS += $$QTFIREBASE_SDK_LIBS_PATH/libapp.a
+LIBS += -L$$QTFIREBASE_SDK_LIBS_PATH -lapp
```
Comment (Larpon) [inline qtfirebase_target.pri]: Very nice!

**Pull request merged**